### DISCUSSION TITLE
Is ts_query_cursor_next_capture supposed to change the match.capture_count?

### DISCUSSION THREAD

**Discussion by andremarianiello** [tree-sitter/tree-sitter#1946] [Q&A]
I filed an issue with neovim (https://github.com/neovim/neovim/issues/20948) because I assume that the observed behavior of the treesitter api is intentional, but I wanted to ask a question here because I couldn't find it documented and I wanted to confirm. Put shortly, I observed that when `ts_query_cursor_next_capture` is used on a query pattern with **2** captures, the `match.capture_count` is **1** for the _first_ capture, and then increases to **2** for the _second_ capture, as if the `capture_count` is counting the number of captures in the capture list, _NOT_ the number of captures in the capture pattern. Is this correct and expected behavior?

Second question: Neovim is currently using capture_count to determine if a pattern match has more than one capture in it, which now seems incorrect. Is there a better way to make that determination using the existing api?

Thanks!

**Comment by ahlinc** [comment]
> the `match.capture_count` is **1** for the _first_ capture, and then increases to **2** for the _second_ capture

Sound like a weird behavior and looks like a bug.  I'll check it on occasion.
